### DISCUSSION TITLE
Feature Testing

### DISCUSSION THREAD

**Discussion by kingdomac** [laravel/framework#39764] [Q&A]
- Laravel Version: 8.70.2
- PHP Version: 8.0.2
- Database Driver & Version: 10.4.17-MariaDB

### Description:
In my test classes, I'm creating records inside my database (using LazilyRefreshDatabase) and sending requests to the endpoint of my API, so when running the test:
- succeed: when the test runs independently for each function 
- Fails:  when running the whole testing classes or even for one class if it includes test cases which create new records for the same table.

### Steps To Reproduce:
The problem is that in some testing functions I'm creating a number of records for the same table in each function using the factory.
In my controller function, I have set a validation rule to check for an array of IDs if it exists in the database and when referring for some records using the array[0] or array[1] (in my test functions) and sending to the API endpoint,  when the whole test run the response returns the validation error: ""errors": {
        "items.0": [
            "The selected items.0 is invalid."
        ],
        "items.1": [
            "The selected items1 is invalid."
        ]
    }"

In my test function:
$products = Product::factory(5)->create();
// I am creating other products in the same class and other testing classes

$order = [
            'name' => $this->faker->name(),
            'email' => $this->faker->email(),
            'address' => $this->faker->address(),
            'paymentMethod' => PayPalService::PAYMENT_METHOD,
            'items' => [
                ['id' => $products[0]->id, 'quantity' => 2],
                ['id' => $products[1]->id, 'quantity' => 1]
            ]
        ];

The validation rule is : 'items.*' => [Rule::exists('products', 'id')],
It seems it is related to the state of the database when refreshing, some other testing classes or functions are changing the state which deletes other states and gives like the created records in the factory doesn't exist anymore.

**Comment by jyrkidn** [comment]
Hey @kingdomac, we also stumbled upon this issue. After a lot of debugging, we fixed it by using `'exists:App\Models\User,id'` instead of `'exists:users,id'`.

Hope this helps someone else in the future.

**Reply by kingdomac** [reply]
It is the same as the User model by default points to the users table. Both works fine.

**Reply by kingdomac** [reply]
It has been solved, the issue was unstead of items.* It should be items.*.id as i remember in the validation rules.
